django_project/minisass_authentication/views/yoma_auth.py

```diff
@@ -0,0 +1,320 @@
+                return Response(token_response, status=status.HTTP_400_BAD_REQUEST)
+
+            # Decode the ID token to get user information
+            try:
+                # Note: In production, you should verify the JWT signature
+                # For now, we'll decode without verification (not recommended for production)
+                id_token = token_response.get('id_token')
+                if not id_token:
+                    return Response(
+                        {'error': 'missing_id_token', 'error_description': 'ID token not found in response'},
+                        status=status.HTTP_400_BAD_REQUEST
+                    )
+                # Decode JWT without verification (add verification in production)
+                decoded_token = jwt.decode(id_token, options={"verify_signature": False})
+
+                # Extract user information from the token
+                email = decoded_token.get('email')
+                username = decoded_token.get('preferred_username') or decoded_token.get('sub')
+                first_name = decoded_token.get('given_name', '')
+                last_name = decoded_token.get('family_name', '')
+
+                if not email or not username:
+                    return Response(
+                        {'error': 'invalid_token', 'error_description': 'Required user information not found in token'},
+                        status=status.HTTP_400_BAD_REQUEST
+                    )
+
+                # Generate a complex random password
+                def generate_complex_password(length=16):
+                    """Generate a complex random password with uppercase, lowercase, digits, and special characters."""
+                    # Ensure at least one character from each category
+                    uppercase = secrets.choice(string.ascii_uppercase)
+                    lowercase = secrets.choice(string.ascii_lowercase)
+                    digit = secrets.choice(string.digits)
+                    special = secrets.choice('!@#$%^&*()_+-=[]{}|;:,.<>?')
+
+                    # Fill the rest with random characters from all categories
+                    all_chars = string.ascii_letters + string.digits + '!@#$%^&*()_+-=[]{}|;:,.<>?'
+                    remaining = ''.join(secrets.choice(all_chars) for _ in range(length - 4))
+
+                    # Combine and shuffle
+                    password_chars = list(uppercase + lowercase + digit + special + remaining)
+                    secrets.SystemRandom().shuffle(password_chars)
+
+                    return ''.join(password_chars)
+
+                # Get or create user
+                user, created = User.objects.get_or_create(
+                    email=email,
+                    defaults={
+                        'username': username,
+                        'first_name': first_name,
+                        'last_name': last_name,
+                        'password': make_password(generate_complex_password()),
```

Comment:
@zamuzakki I don't think this is necessary, since the password will be unusable. Could you set it to empty instead, and mark the user as having unusable password : https://docs.djangoproject.com/en/5.2/ref/contrib/auth/#django.contrib.auth.models.User.set_unusable_password